windows-gaming-driver/src/driver/control/codec.rs

```diff
@@ -50,3 +52,40 @@ impl Encoder for Codec {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bytes::BytesMut;
+    use tokio_io::codec::Decoder;
+
+    macro_rules! test {
+        (fn $name:ident, with $data:expr, expect $expected:expr, len $len:expr) => (
```

Comment:
Do you think that 

```rust
test!(none, &[], None, 0);
test!(invalid, &[0], None, 1);
test!(io_entry, &[1], Some(ControlCmdIn::IoEntry), 0);
```

is easier to read and understand? I personally find it better to read the current version as it is clearer.

Regarding your keywords are not used argument: Especially in libraries I rarely see macros which don't use keywords to indicate usage.